File: crossecore-csharp/Java/Java/Java_/ConditionalExpressionBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class ConditionalExpressionBase 
	:ExpressionImpl, ConditionalExpression
	{
		private Expression _elseExpression;
		public virtual Expression elseExpression
		{
			get {
			
				return _elseExpression;
			}
			set {
				if (value != _elseExpression) {
					NotificationChain msgs = null;
					if (_elseExpression != null){
						msgs = ((InternalEObject)_elseExpression).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CONDITIONALEXPRESSION_ELSEEXPRESSION, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CONDITIONALEXPRESSION_ELSEEXPRESSION, null, msgs);
					}
					msgs = basicSetElseExpression(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.CONDITIONALEXPRESSION_ELSEEXPRESSION , value, value));
				}
				}
		}
		private Expression _thenExpression;
		public virtual Expression thenExpression
		{
			get {
			
				return _thenExpression;
			}
			set {
				if (value != _thenExpression) {
					NotificationChain msgs = null;
					if (_thenExpression != null){
						msgs = ((InternalEObject)_thenExpression).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CONDITIONALEXPRESSION_THENEXPRESSION, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CONDITIONALEXPRESSION_THENEXPRESSION, null, msgs);
					}
					msgs = basicSetThenExpression(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.CONDITIONALEXPRESSION_THENEXPRESSION , value, value));
				}
				}
		}
		private Expression _expression;
		public virtual Expression expression
		{
			get {
			
				return _expression;
			}
			set {
				if (value != _expression) {
					NotificationChain msgs = null;
					if (_expression != null){
						msgs = ((InternalEObject)_expression).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CONDITIONALEXPRESSION_EXPRESSION, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CONDITIONALEXPRESSION_EXPRESSION, null, msgs);
					}
					msgs = basicSetExpression(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.CONDITIONALEXPRESSION_EXPRESSION , value, value));
				}
				}
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.CONDITIONALEXPRESSION;
		}
		
		
		public NotificationChain basicSetElseExpression(Expression newobj, NotificationChain msgs) {
			var oldobj = _elseExpression;
			_elseExpression = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.CONDITIONALEXPRESSION_ELSEEXPRESSION, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		public NotificationChain basicSetThenExpression(Expression newobj, NotificationChain msgs) {
			var oldobj = _thenExpression;
			_thenExpression = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.CONDITIONALEXPRESSION_THENEXPRESSION, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		public NotificationChain basicSetExpression(Expression newobj, NotificationChain msgs) {
			var oldobj = _expression;
			_expression = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.CONDITIONALEXPRESSION_EXPRESSION, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.CONDITIONALEXPRESSION_COMMENTS:
					return comments;
				case Java_PackageImpl.CONDITIONALEXPRESSION_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.CONDITIONALEXPRESSION_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.CONDITIONALEXPRESSION_ELSEEXPRESSION:
					return elseExpression;
				case Java_PackageImpl.CONDITIONALEXPRESSION_EXPRESSION:
					return expression;
				case Java_PackageImpl.CONDITIONALEXPRESSION_THENEXPRESSION:
					return thenExpression;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.CONDITIONALEXPRESSION_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.CONDITIONALEXPRESSION_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.CONDITIONALEXPRESSION_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.CONDITIONALEXPRESSION_ELSEEXPRESSION:
					elseExpression = (Expression) newValue;
					return;
				case Java_PackageImpl.CONDITIONALEXPRESSION_EXPRESSION:
					expression = (Expression) newValue;
					return;
				case Java_PackageImpl.CONDITIONALEXPRESSION_THENEXPRESSION:
					thenExpression = (Expression) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/ConditionalExpressionBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ExpressionImpl} from "./ExpressionImpl";
import {Expression} from "./Expression";
import {ConditionalExpression} from "./ConditionalExpression";
		
			export class ConditionalExpressionBase
			extends ExpressionImpl
			implements ConditionalExpression
			{
				private _elseExpression:Expression = null;
				get elseExpression():Expression{
				
					return this._elseExpression;
				}
				set elseExpression(value:Expression) {
					if (value != this._elseExpression) {
						let msgs:NotificationChain = null;
						if (this._elseExpression != null){
							msgs = (this._elseExpression).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CONDITIONALEXPRESSION_ELSEEXPRESSION, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CONDITIONALEXPRESSION_ELSEEXPRESSION, /*null*/ null, msgs);
						}
						msgs = this.basicSetElseExpression(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.CONDITIONALEXPRESSION_ELSEEXPRESSION , value, value));
					}
				}
				private _expression:Expression = null;
				get expression():Expression{
				
					return this._expression;
				}
				set expression(value:Expression) {
					if (value != this._expression) {
						let msgs:NotificationChain = null;
						if (this._expression != null){
							msgs = (this._expression).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CONDITIONALEXPRESSION_EXPRESSION, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CONDITIONALEXPRESSION_EXPRESSION, /*null*/ null, msgs);
						}
						msgs = this.basicSetExpression(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.CONDITIONALEXPRESSION_EXPRESSION , value, value));
					}
				}
				private _thenExpression:Expression = null;
				get thenExpression():Expression{
				
					return this._thenExpression;
				}
				set thenExpression(value:Expression) {
					if (value != this._thenExpression) {
						let msgs:NotificationChain = null;
						if (this._thenExpression != null){
							msgs = (this._thenExpression).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CONDITIONALEXPRESSION_THENEXPRESSION, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CONDITIONALEXPRESSION_THENEXPRESSION, /*null*/ null, msgs);
						}
						msgs = this.basicSetThenExpression(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.CONDITIONALEXPRESSION_THENEXPRESSION , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ConditionalExpressionBase.eStaticClass;
				}
			
			
				public basicSetElseExpression(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._elseExpression;
					this._elseExpression = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.CONDITIONALEXPRESSION_ELSEEXPRESSION, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				public basicSetExpression(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._expression;
					this._expression = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.CONDITIONALEXPRESSION_EXPRESSION, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				public basicSetThenExpression(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._thenExpression;
					this._thenExpression = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.CONDITIONALEXPRESSION_THENEXPRESSION, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.CONDITIONALEXPRESSION_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_ELSEEXPRESSION:
							return this.elseExpression;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_EXPRESSION:
							return this.expression;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_THENEXPRESSION:
							return this.thenExpression;
					}
					//return this.eGetFromExpression(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.CONDITIONALEXPRESSION_ELSEEXPRESSION:
							this.elseExpression = <Expression> newValue;
							return;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_EXPRESSION:
							this.expression = <Expression> newValue;
							return;
						case Java_PackageLiterals.CONDITIONALEXPRESSION_THENEXPRESSION:
							this.thenExpression = <Expression> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
